### PR TITLE
Support glam and macaw types

### PR DESCRIPTION
**Pull request by davidpdrsn**
### Checklist

* [x] I have read the [Contributor Guide](../../CONTRIBUTING.md)
* [x] I have read and agree to the [Code of Conduct](../../CODE_OF_CONDUCT.md)
* [x] I have added a description of my changes and why I'd like them included in the section below

### Description of Changes

This adds impls for `Reflect` and friends for types from glam and macaw, behind a feature flag.

### CODE REVIEW

**Comment by davidpdrsn** [comment]
Can't quite merge this yet. We need some special handling of glam's types that changes depending on architecture (such as `Vec4`).

**Comment by davidpdrsn** [comment]
This should be good to go now. I've tested compiling it with and without simd support for glam (using its `core-simd` feature) and it works. I just decided not to support the types that don't have a stable definition for now.

File: crates/mirror-mirror/src/foreign_impls/mod.rs [hunk 1]
```diff
@@ -80,3 +80,52 @@ __private_derive_reflect_foreign! {
         end: Idx,
     }
 }
+
+#[cfg(feature = "glam")]
+mod glam_impls {
```
Comment (bnjbvr) [inline crates/mirror-mirror/src/foreign_impls/mod.rs:85]: Would it make sense to put this in an `glam.rs` file? It's a bit weird to have it in the `mod.rs` module here (I did look for it in another file when trying that PR locally.)

File: crates/mirror-mirror/src/foreign_impls/mod.rs [hunk 2]
```diff
@@ -80,3 +80,52 @@ __private_derive_reflect_foreign! {
...
+        pub struct Mat3 {
+            pub x_axis: Vec3,
+            pub y_axis: Vec3,
+            pub z_axis: Vec3,
+        }
+    }
+}
+
+#[cfg(feature = "macaw")]
+mod macaw_impls {
```
Comment (bnjbvr) [inline crates/mirror-mirror/src/foreign_impls/mod.rs:123]: ditto here; even if it makes for a tiny file, it's nice to have it separate from the rest of the core types imo

**Review by bnjbvr — APPROVED** [review]
Makes sense, thanks! Deref magic in glam is magic.

**Comment by bnjbvr** [comment]
Hah, auto-merge is fast! I'll open a PR with the suggested changes.

**Pull request merged**